### PR TITLE
install envoy gateway using helm tool

### PR DESCRIPTION
**Pull request by eguzki**

### CODE REVIEW

File: make/envoy-gateway.mk [hunk 1]
```diff
@@ -8,31 +8,28 @@ EG_NAMESPACE = envoy-gateway-system
...
 	-rm -rf $(TMP)
 
 .PHONY: egctl
 egctl: $(EGCTL) ## Download egctl locally if necessary.
 
+EG_VERSION ?= v0.0.0-latest
 .PHONY: envoy-gateway-install
 envoy-gateway-install: kustomize
-	$(KUSTOMIZE) build $(EG_CONFIG_DIR) | kubectl apply -f -
+	$(HELM) install eg oci://docker.io/envoyproxy/gateway-helm --version $(EG_VERSION) -n envoy-gateway-system --create-namespace
```
Comment (adam-cattermole) [inline make/envoy-gateway.mk:30]: `EG_VERSION` was replaced by `EGCTL_VERSION`
```suggestion
	$(HELM) install eg oci://docker.io/envoyproxy/gateway-helm --version $(EGCTL_VERSION) -n envoy-gateway-system --create-namespace
```

Comment (eguzki) [reply make/envoy-gateway.mk:30]: it should be `EG_VERSION`.

File: make/envoy-gateway.mk [hunk 2]
```diff
@@ -8,31 +8,28 @@ EG_NAMESPACE = envoy-gateway-system
...
-	cd $(TMP); curl -sSL https://github.com/envoyproxy/gateway/releases/download/$(EG_VERSION)/egctl_$(EG_VERSION)_$(OS)_$(ARCH).tar.gz -o egctl.tar.gz
+	cd $(TMP); curl -sSL https://github.com/envoyproxy/gateway/releases/download/$(EGCTL_VERSION)/egctl_$(EGCTL_VERSION)_$(OS)_$(ARCH).tar.gz -o egctl.tar.gz
 	tar xf $(TMP)/egctl.tar.gz -C $(TMP)
 	cp $(TMP)/bin/$(OS)/$(ARCH)/egctl $(EGCTL)
 	-rm -rf $(TMP)
 
 .PHONY: egctl
 egctl: $(EGCTL) ## Download egctl locally if necessary.
 
+EG_VERSION ?= v0.0.0-latest
```
Comment (eguzki) [inline make/envoy-gateway.mk:27]: heads up as the default version is the `latest`!!

for v1.0.0 we should be running 

```
make local-env-setup EG_VERSION=v1.0.0
```

File: make/envoy-gateway.mk (hunk 1 above)
Comment (adam-cattermole) [reply make/envoy-gateway.mk:30]: Ahh I see it was set below, makes sense!

**Review by adam-cattermole — APPROVED** [review]

**Pull request merged**